oio/xcute/common/worker.py

```diff
@@ -29,22 +29,19 @@ def __init__(self, conf, logger=None):
         self.conf = conf
         self.logger = logger or get_logger(self.conf)
         self.beanstalkd_senders = dict()
-        self.jobs = CacheDict(size=10)
+        self.tasks = CacheDict(size=10)
 
     def process_beanstalkd_job(self, beanstalkd_job):
         job_id = beanstalkd_job['job_id']
 
-        job = self.jobs.get(job_id)
-        if job is None:
+        task = self.tasks.get(job_id)
+        if task is None:
             job_type = beanstalkd_job['job_type']
-            job_class = JOB_TYPES[job_type]
+            task_class = JOB_TYPES[job_type].TASK_CLASS
             job_config = beanstalkd_job['job_config']
 
-            job = job_class(self.conf, logger=self.logger)
-            job.load_config(job_config)
-            job.init_process_task()
-
-            self.jobs[job_id] = job
+            task = task_class(self.conf, job_config, logger=self.logger)
```

Comment:
I still think the task doesn't need to be passed all of the job config, but only its specific parameters.